pkg/controllers/leaderworkerset_controller.go

```diff
@@ -274,11 +293,16 @@ func (r *LeaderWorkerSetReconciler) rollingUpdateParameters(ctx context.Context,
 	return min(partition, utils.NonZeroValue(stsReplicas-int32(rollingStep)-continuousReadyReplicas)), wantReplicas(lwsUnreadyReplicas), nil
 }
 
-func (r *LeaderWorkerSetReconciler) SSAWithStatefulset(ctx context.Context, lws *leaderworkerset.LeaderWorkerSet, partition, replicas int32) error {
+func (r *LeaderWorkerSetReconciler) SSAWithStatefulset(ctx context.Context, lws *leaderworkerset.LeaderWorkerSet, partition, replicas int32, templateHash string) error {
 	log := ctrl.LoggerFrom(ctx)
 
+	if err := revisionutils.CreateLeaderWorkerSetRevision(ctx, r.Client, lws, templateHash); err != nil {
+		log.Error(err, "Creating LWS Revision")
+		return err
+	}
```

Comment:
We shouldn't be creating here, the creation will be done at the beginning of the reconcile function via `createControllerRevisionIfNonExist`

Reply:
Removing this one + removing `templateHash` requires changes to the way we set the `templateHash` label in the revision. With these changes, we rely on leader sts to determine what templateHash to use. That is why `createControllerRevisionIfNonExist` checks if the leaderSts exists before attempting to create a new one. When a leaderSts doesn't exist however, we need to determine what templateHash to use. But the templateHash depends on a revision being created. 

So in the case where the leaderSts doesn't exist yet, we would need to set the value of `templateHash` to the hashed controller revision, basically having two labels with the same value. 

Or, we can change the `controller.kubernetes.io/hash` label to be our selector instead. In the case where we pass a templateHash, we set that label equal to the templateHash value.